cEP-0031.md

```diff
@@ -0,0 +1,254 @@
+| -------- | ------------------------------------------- |
+| cEP      | 31                                          |
+| Version  | 0.1                                         |
+| Title    | Improve Generic Bear Quality                |
+| Authors  | Bhushan Khanale <mailto:bkhanale@gmail.com> |
+| Status   | Proposed                                    |
+| Type     | Feature                                     |
+
+## Abstract
+
+This cEP describes the improvement in the generic bears currently available
+along with the implementation of the new bears as a part of the
+[GSoC 2019 Project][project].
+
+## Introduction
+
+coala has few generic bears which have the potential to perform better
+fixing some of the issues in them. Some of these issues are as follows:
+
+-   [coala/coala-bears#2092][2092]: Pygmented AnnotationBear
+    Pygments is a Python library for syntax highlighting of source code.
+    Therefore it contains complete syntax definitions of any relevant
+    programming language and a custom parser framework for tokenizing source
+    code of all those languages. This gives the opportunity of reducing the
+    current complex implementation of the bear into a much simpler one.
+
+-   [coala/coala-bears#644][644]: Ignore doc comments/doc strings
+    Comments or docstrings do not have any set standards while the
+    IndentationBear does not ignore them in a file. IndentationBear checks for
+    correct indentation in the program statements since that is crucial for
+    proper code style. Docstrings/comments, on the other hand, do not have any
+    such compulsion. So ideally docstrings should be ignored by the bear. coala
+    already uses documentation extraction algorithms in DocumentationStyleBear.
+    Using the existing algorithms and constructing ignore ranges out of the
+    extracted comments inside the IndentationBear, this issue can be solved.
+
+-   [coala/coala-bears#502][502]: MarkdownBear is not very helpful
+    On running MarkdownBear all we get is the result message 'The text does not
+    comply to the set style.'. This is not very helpful for most users. We will
+    be using a plugin to solve this issue.
+
+-   [coala/coala-bears#1897][1897]: Show PEP8 error description
+    On running PEP8Bear the default message is 'The code does not comply to
+    PEP8.' which is not very helpful for the user. This issue will be solved
+    by implementing upstream changes to autopep8.
+
+-   [hhatto/autopep8#227][227]: Create diff for a specific pep8 issue
+    User should have a choice of selecting the issue he wants to generate a fix
+    for. Although the issue is a part of the autopep8, we will be solving
+    this as a part of upstream changes to autopep8.
+
+Along with fixing these issues, new bears including `OutdatedDependencyBear`,
+`FileExistsBear`, `RegexLintBear`, `FileModeBear` and `RequirementsCheckBear`
+will be implemented.
+
+## Implementation of new bears
+
+### 1. OutdatedDependencyBear
```

Comment:
Dependency management is done in https://gitlab.com/coala/package_manager/

This was mentioned at https://github.com/coala/coala-bears/issues/2445#issuecomment-432134963

We will not have a new bear in the coala-bears repo which attempts to reproduce all of this.

How will you CI it all?  The coala-bears CI already breaks all of the time.

Please redesign your bear with the implementation using one of the recommended solutions in https://github.com/coala/coala-bears/issues/2445#issuecomment-432134963

Reply:
I'm introducing changes in the package_manager itself, the code below belongs there. I will be using them in this bear.

Reply:
That is not clear.  Explicitly mention which repo requires enhancement.  And include in your cEP a non-pip example -- think about a hard language/manager which doesnt have all the features of pip.  Go?

More-over you have not included here any details about how the bear would work.  More details!  Then you can get more feedback on your design.

Reply:
Details coming up.